api/assessment/assessment.go

```diff
@@ -30,17 +30,21 @@ import "errors"
 type ResultHookFunc func(result *AssessmentResult, err error)
 
 var (
+	ErrIdMissing                             = errors.New("id in assessment result is missing")
 	ErrTimestampMissing                      = errors.New("timestamp in assessment result is missing")
 	ErrMetricIdMissing                       = errors.New("metric id in assessment result is missing")
 	ErrMetricConfigurationMissing            = errors.New("metric configuration in assessment result is missing")
 	ErrEvidenceIdMissing                     = errors.New("evidence id in assessment result is missing")
-	ErrNonComplianceCommentsMissing          = errors.New("non-compliance comments in assessment result is missing")
 	ErrMetricConfigurationOperatorMissing    = errors.New("operator in metric data is missing")
 	ErrMetricConfigurationTargetValueMissing = errors.New("target value in metric data is missing")
 )
 
 // Validate validates the assessment result according to several required fields
 func (result *AssessmentResult) Validate() (resourceId string, err error) {
+	if result.Id == "" {
```

Comment:
Should we also check if the id is a UUID? Or do you think this is too performance-heavy? Also potentially for the evidence ID.

Reply:
Do we already know in which format the ID should be? If yes, then I would check that.

Reply:
Good point. We could use the error of `uuid.parse(result.Id)` or do a sanity check by just compare the length (36) via `len(result.Id)`. I did a small benchmark test for the 3 options:

0.0000004 ns/op for simple length check
0.0000009 ns/op for empty string check
0.0000050 ns/op for uuid.parse

Since length check is even faster (or at least somehow equally fast) than empty string check (does this makes sense, anyway :D), it wouldn't add significant time. The UUID approach is also not that slow but needs a dependency.

Reply:
We already have GitHub.com/google/uuid as a dependency

Reply:
Then I would just substitute the empty string check with uuid.Parse() way since it also throws error for empty string ("invalid UUID length: 0")